igor-web/src/main/groovy/com/netflix/spinnaker/igor/config/IgorConfig.groovy

```diff
@@ -69,6 +70,11 @@ class IgorConfig extends WebMvcConfigurerAdapter {
         new BuildServices()
     }
 
+    @Bean
```

Comment:
Curious about the decision to explicitly create a `@Bean` rather than add a `@Component` annotation on the class...I see examples of both in our codebase, and sometimes wonder which one to do.  (Entirely just a question out of curiosity, not a suggestion to change this.)

Reply:
For here specifically I followed what you did for `BuildServices` 😄. In general, I'll do `@Component` if that's what everything in the module does. If some are defined with `@Bean` in the config file I'll do that so that everything is in one place. Also, if something is designed to be replaced by a custom component (and therefore needs the `@ConditionalOnMissingBean`) you can only do that with the `@Bean` way. Sometimes if there are `@ConditionalOnProperty` or two services that are turned on/off by one property value I'll put them together in a config so that you can clearly see what the property controls.

Hopefully that helps with how I think about things!

Reply:
> `@Component` (and `@Service` and `@Repository`) are used to auto-detect and auto-configure beans using classpath scanning. There's an implicit one-to-one mapping between the annotated class and the bean (i.e. one bean per class). Control of wiring is quite limited with this approach, since it's purely declarative.

> `@Bean` is used to explicitly declare a single bean, rather than letting Spring do it automatically as above. It decouples the declaration of the bean from the class definition, and lets you create and configure beans exactly how you choose.

https://stackoverflow.com/questions/10604298/spring-component-versus-bean

Reply:
Yup, that makes sense, thanks!